collect_app/src/main/java/org/odk/collect/android/utilities/MediaUtils.java

```diff
@@ -476,12 +476,11 @@ public static String getPath(final Context context, final Uri uri) {
                 final String[] split = docId.split(":");
                 final String type = split[0];
 
-                if ("primary".equalsIgnoreCase(type)) {
-                    return Environment.getExternalStorageDirectory() + "/"
-                            + split[1];
```

Comment:
I experimentally confirmed that `return Environment.getExternalStorageDirectory() + "/" + split[1];`  works not only for primary type. Since we don't have any fallback for non-primary volumes (see removed TODO below), even just removing of wrapping `if ("primary".equalsIgnoreCase(type))` condition will significantly reduce number of devices affected by the issue.  

`Environment.getExternalStorageDirectory()` usually returns value like `/storage/sdcard/`, which works for most of cases I've tested.
 
On some devices path can look like `/storage/4A50-B543/path/to/fils/on/sd/card/file.jpg`, where `4A50-B543` is the value of `type` variable, parsed from the initial URI.  `Environment.getExternalStorageDirectory()` doesn't return valid value for such case and there is no utility method which returns/builds such path. 
So I decided indicate such devices by checking that `new File("/storage/4A50-B543").exists()` and build path manually.